src/modules/article/article.controller.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  Controller,
+  Delete,
+  Get,
+  HttpStatus,
+  Post,
+  Put, Query, UsePipes,
+} from '@nestjs/common';
+import { ApiResponse, ApiTags } from '@nestjs/swagger';
+import { ArticleService } from './article.service';
+import { ROUTES } from '../../shared/config/routes';
+import { GetByGenderAndCategoryDto } from './dto/received/get-by-gender-and-category.dto';
+import { ValidationPipe } from '../../shared/pipes/validation.pipe';
+import { GetByGenderDto } from './dto/received/get-by-gender.dto';
+import { GetByCategoryDto } from './dto/received/get-by-category.dto';
+import { GetByIdDto } from '../../shared/dto/get-by-id.dto';
+import { CreateArticleDto } from './dto/received/create-article.dto';
+import { ArticleDto } from '../database/dto/article.dto';
+import { UpdateArticleDto } from './dto/received/update-article.dto';
+
+@ApiTags(ROUTES.ARTICLE.MAIN)
+@Controller(ROUTES.ARTICLE.MAIN)
+export class ArticleController {
+  constructor(private readonly articleService: ArticleService) {
+  }
+
+  @Get()
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'Articles was found',
+    type: ArticleDto,
+  })
+  @UsePipes(new ValidationPipe())
+  getArticles(): Promise<ArticleDto[]> {
+    return this.articleService.getAll();
+  }
+
+  @Get(ROUTES.ID.DYNAMIC_ID)
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'Article was found',
+    type: ArticleDto,
+  })
+  @UsePipes(new ValidationPipe())
+  getArticleById(@Query() query: GetByIdDto): Promise<ArticleDto> {
+    return this.articleService.getById(query.id);
+  }
+
+  @Get(ROUTES.ID.DYNAMIC_ID)
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'Articles was found',
+    type: ArticleDto,
+  })
+  @UsePipes(new ValidationPipe())
+  getArticlesByUserId(@Query() query: GetByIdDto): Promise<ArticleDto[]> {
+    return this.articleService.getArticlesByUserId(query.id);
+  }
+
+  @Post()
+  @ApiResponse({
+    status: HttpStatus.CREATED,
+    description: 'Article created',
+    type: ArticleDto,
+  })
+  @UsePipes(new ValidationPipe())
+  createArticle(@Query() query: CreateArticleDto): Promise<ArticleDto> {
+    return this.articleService.createOne(query.article);
+  }
+
+  @Put(ROUTES.ID.DYNAMIC_ID)
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'Article updated',
+    type: ArticleDto,
+  })
+  @UsePipes(new ValidationPipe())
+  updateArticle(@Query() query: UpdateArticleDto): Promise<ArticleDto> {
+    return this.articleService.update(query.articleId, query.article);
+  }
+
+  @Delete(ROUTES.ID.DYNAMIC_ID)
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'Article deleted',
+    type: ArticleDto,
+  })
+  @UsePipes(new ValidationPipe())
+  deleteArticle(@Query() query: GetByIdDto): Promise<ArticleDto> {
+    return this.articleService.remove(query.id);
+  }
```

Comment:
you put param flag in Delete decorator, and then wait id in query?!?!?! if you add :id - you need to wait data from @Params